ui/src/app/utils/getStatusText.ts

```diff
@@ -0,0 +1,23 @@
+import { nodeStatus } from "app/base/enum";
+import type { Host } from "app/store/types/host";
+
+// Node statuses for which the OS + release is made human-readable.
+const formattedReleaseStatuses = [nodeStatus.DEPLOYED, nodeStatus.DEPLOYING];
+
+/**
+ * Returns formatted status text of a given node.
+ * @param node - the node who's status you are checking.
+ * @param release - the formatted release of the node.
+ * @returns formatted status text
+ */
+export const getStatusText = (node: Host | null, release: string): string => {
+  if (!node) {
+    return "Unknown";
+  }
+  if (release && formattedReleaseStatuses.includes(node.status_code)) {
+    return node.status_code === nodeStatus.DEPLOYING
```

Comment:
I know you didn't add this in this branch, but should this check for `show_os_info` like it does on machine details? I'm not quite sure what that param is there for...

Reply:
Yeah it's kinda strange but `show_os_info` is machine details-specific. In the machine list we only show the OS if the machine is deploying or deployed, but in machine details it's showed for a few extra statuses. I guess it's because machine details has a bit of extra real estate to show the status AND the OS.